crates/ruff_python_semantic/src/analyze/typing.rs

```diff
@@ -418,6 +418,49 @@ where
     inner(func, semantic, expr, None);
 }
 
+/// Traverse a "literal" type annotation, applying `func` to each literal member.
+///
+/// The function is called with each expression in the literal (excluding declarations of nested
+/// literals) and the parent expression.
+pub fn traverse_literal<'a, F>(func: &mut F, semantic: &SemanticModel, expr: &'a Expr)
+where
+    F: FnMut(&'a Expr, &'a Expr),
+{
+    fn inner<'a, F>(
+        func: &mut F,
+        semantic: &SemanticModel,
+        expr: &'a Expr,
+        parent: Option<&'a Expr>,
+    ) where
+        F: FnMut(&'a Expr, &'a Expr),
+    {
+        // Ex) Literal[x, y]
+        if let Expr::Subscript(ast::ExprSubscript { value, slice, .. }) = expr {
+            if semantic.match_typing_expr(value, "Literal") {
+                match slice.as_ref() {
```

Comment:
nit: I know the existing `traverse_union` function in this file uses `.as_ref()` here, but it's best to avoid `AsRef<>` where possible in situations like this. A single struct in Ruff can have multiple implementations of the `AsRef` trait, so if you use `.as_ref()`, you're relying on type inference by the compiler to ensure that it gets converted to the correct type. That's potentially fragile -- it could unexpectedly break if the struct gets more `AsRef` implementations added in the future.

`&**slice` means:
- Take the `slice` variable, which currently has type `&Box<Expr>`
- Dereference it once: -> `Box<Expr>`
- Dereference it again: -> `Expr`
- And now borrow it: -> `&Expr`

```suggestion
                match &**slice {
```

Reply:
Clippy has suggested me &** before, but it seemed awkward, it's not used very much in the codebase so I refrained.

I'll take care of the nits, and also have a PR up for `PYI064` in a couple hours.

Reply:
We're having this discussion internally now :)

Reply:
Yeah, feel free to leave this for now if you don't like it :-)

I personally like `&**` as I feel like it's more explicit about exactly what you want to happen to the type. But as you say, we're pretty inconsistent about this currently, and it's not a big deal either way :-)